### ISSUE TITLE
Visual Studio 2017 Not Found

### ISSUE THREAD

**Issue by isaac2004** [Microsoft/vsts-tasks#4361]
Trying to Build a .Net 4.5.2 Web Project and getting this error

> 2017-05-19T13:59:52.5863673Z ==============================================================================
2017-05-19T13:59:52.5863673Z Task         : Visual Studio Build
2017-05-19T13:59:52.5863673Z Description  : Build with MSBuild and set the Visual Studio version property
2017-05-19T13:59:52.5863673Z Version      : 1.117.0
2017-05-19T13:59:52.5863673Z Author       : Microsoft Corporation
2017-05-19T13:59:52.5863673Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613727)
2017-05-19T13:59:52.5863673Z ==============================================================================
2017-05-19T14:00:00.3322592Z ##[command]"D:\a\_tasks\VSBuild_71a9a2d3-a98a-4caa-96ab-affca411ecda\1.117.0\ps_modules\MSBuildHelpers\vswhere.exe" -version [15.0,15.1) -latest -format json
2017-05-19T14:00:00.7462596Z ##[error]Visual Studio 2017 was not found.
2017-05-19T14:00:00.8124660Z ##[section]Finishing: Build solution

**Comment by bryanmacfarlane** [comment]
Are you trying to build a Visual studio 2017 solution?  Are you using hosted or private agent?  Did you select the VS2017 queue on the options tab?  (that points to machines with vs2017)

**Comment by isaac2004** [comment]
This is a 2017 Solution, hosted. Changing the queue fixed this issue

**Comment by bryanmacfarlane** [comment]
Yeah - this USA bug usability problem for us.  We're aware and discussing

**Comment by bryanmacfarlane** [comment]
Is a ...

**Comment by alexxonline** [comment]
What is the workaround for this issue?

**Comment by isaac2004** [comment]
@alexxonline there is an option in the Agent Section of Options. Set the Default Agent Queue to Hosted VS 2017

![image](https://cloud.githubusercontent.com/assets/8878502/26364005/da9f2730-3fb0-11e7-9fdd-3dd4c1c67ed2.png)

**Comment by bryanmacfarlane** [comment]
And to be clear that is the solution, not a workaround.  The usability issue I'm referring to is it's not obvious you have to do that

**Comment by styfle** [comment]
@isaac2004 Thank you, this worked! 

@bryanmacfarlane The odd part is that I switched the build from VS 2015 to VS 2017 in mid April and it worked fine. This last week starting mid May is when I started getting the "Visual Studio 2017 was not found." error.

After changing the agent queue from Default to Hosted VS2017, I can build but not run tests.

### Console Output

```
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3276: Found conflicts between different versions of the same dependent assembly. Please set the "AutoGenerateBindingRedirects" property to true in the project file. For more information, see http://go.microsoft.com/fwlink/?LinkId=294190. [d:\a\1\s\Core Projects\SearchDataManager\SearchDataManager.csproj]
    115 Warning(s)
    0 Error(s)
Time Elapsed 00:00:31.01
******************************************************************************
Finishing: Build solution Core Projects/MySolution.sln
******************************************************************************
******************************************************************************
Starting: Test Assemblies **\*test*.dll;-:**\obj\**
******************************************************************************
==============================================================================
Task         : Visual Studio Test
Description  : Run tests with Visual Studio test runner
Version      : 1.0.85
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=624539)
==============================================================================
Preparing task execution handler.
Executing the powershell script: d:\a\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\1.0.85\VSTest.ps1
Install Visual Studio 2015 Update 1 or higher on your build agent machine to run the tests in parallel.
No results found to publish.
System.Management.Automation.CmdletInvocationException: Unable to determine the location of vstest.console.exe ---> System.IO.FileNotFoundException: Unable to determine the location of vstest.console.exe
   at Microsoft.TeamFoundation.DistributedTask.Task.Internal.InvokeVSTestCmdlet.GetVsTestLocation()
   at Microsoft.TeamFoundation.DistributedTask.Task.Internal.InvokeVSTestCmdlet.ProcessRecord()
   at System.Management.Automation.CommandProcessor.ProcessRecord()
   --- End of inner exception stack trace ---
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
   at Microsoft.TeamFoundation.DistributedTask.Handlers.LegacyVSTSPowerShellHost.VSTSPowerShellHost.Main(String[] args)
LegacyVSTSPowerShellHost.exe completed with return code: -1.
******************************************************************************
Finishing: Test Assemblies **\*test*.dll;-:**\obj\**
******************************************************************************
```

Any idea about this? Should I open a new issue?

**Comment by styfle** [comment]
I figured it out. There is an advanced options with VSTest Version.

![image](https://cloud.githubusercontent.com/assets/229881/26406514/7d4daafa-4066-11e7-9ca0-61fd817f269b.png)

I selected "Latest" and it works now.

**Comment by samanax** [comment]
Thanks guys for the build and test settings solutions. They work perfectly and you saved me a lot of time!

**Comment by maheshmysagalla** [comment]
where it's located advanced options with VSTest Version.

**Comment by styfle** [comment]
Today, I started getting failing builds for the Hosted 2017 yet no configuration was changed. The error was the following:

```
Unable to determine the location of vstest.console.exe
```

I found a workaround in https://github.com/Microsoft/vsts-tasks/issues/415 which says to update the Test Task from `1.x` to `2.x` and this seemed to work.

@bryanmacfarlane It's a shame that every couple months, these tasks are breaking 😞

**Comment by AndreyShvidky** [comment]
Specifying v.2.0.55 solves the problem, but brings new one.
Something is wrong with console output encoding:
![default](https://user-images.githubusercontent.com/31736985/31248635-84297132-aa25-11e7-9cb4-6b659873e523.PNG)
while on 1.0.85 everything was fine.

**Comment by TehWardy** [comment]
Hmm ... I'm getting the warning too :( and I have the VSTest version set to latest already.
From the build log ...

> Install Visual Studio 2015 Update 1 or higher on your build agent machine to run the tests in parallel.
